src/server/game/Server/WorldSession.cpp

```diff
@@ -132,7 +133,8 @@ WorldSession::WorldSession(uint32 id, std::string&& name, uint32 battlenetAccoun
     _RBACData(NULL),
     expireTime(60000), // 1 min after socket loss, session is deleted
     forceExit(false),
-    m_currentBankerGUID()
+    m_currentBankerGUID(),
+    _battlePetMgr(Trinity::make_unique<BattlePetMgr>(this))
```

Comment:
Nope, we've discussed it with Shauren and made decision to put it here. It's account wide, there's no better place.